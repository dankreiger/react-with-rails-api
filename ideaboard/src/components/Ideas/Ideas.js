import React, { Component } from 'react';
import { Row, Button, Col } from 'reactstrap';
import axios from 'axios';
import Spinner from '../Spinner/Spinner';
import Idea from '../Idea/Idea';
import { IdeasContainer } from './Ideas.style';
import update from 'immutability-helper';

export default class Ideas extends Component {
  state = { ideas: [], editingIdeaId: null };
  componentDidMount() {
    axios
      .get('http://localhost:3001/api/v1/ideas.json')
      .then(response => {
        this.setState({ ideas: response.data });
      })
      .catch(error => console.log(error));
  }
  addNewIdea = () => {
    axios
      .post('http://localhost:3001/api/v1/ideas', {
        idea: { title: '', body: '' }
      })
      .then(response => {
        console.log(response);
        const ideas = update(this.state.ideas, {
          $splice: [[0, 0, response.data]]
        });
        this.setState({ ideas, editingIdeaId: response.data.id });
      })
      .catch(error => console.log(error));
  };
  render() {
    const { editingIdeaId, ideas } = this.state;
    return (
      <IdeasContainer>
        {ideas.length ? (
          <>
            <Row>
              {ideas.map(idea => (
              <Col xs="12" md="6" key={idea.id}>
                <Idea {...idea} currentlyEditing={editingIdeaId === idea.id} />}
              </Col>
              ))}
            </Row>
            <Row>
              <Col>
                <Button color="primary" onClick={this.addNewIdea}>
                  New Idea
                </Button>
              </Col>
            </Row>
          </>
        ) : (
          <Spinner />
        )}
      </IdeasContainer>
    );
  }
}
