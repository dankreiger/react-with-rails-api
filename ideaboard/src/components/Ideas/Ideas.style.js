import styled from 'styled-components';
import { Container } from 'reactstrap';

export const IdeasContainer = styled(Container)`
  height: calc(100vh - 80px);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`
