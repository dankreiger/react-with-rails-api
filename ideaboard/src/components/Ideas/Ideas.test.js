import React from 'react';
import { shallow } from 'enzyme';
import Ideas from './Ideas';

describe('Idea', () => {
  let ideasComponent;

  beforeEach(() => {
    ideasComponent = shallow(<Ideas />);
  });

  describe('rendering', () => {
    test('renders as expected', () => {
      expect(ideasComponent.length).toBe(1);
      expect(ideasComponent).toMatchSnapshot();
    });
  });
});
