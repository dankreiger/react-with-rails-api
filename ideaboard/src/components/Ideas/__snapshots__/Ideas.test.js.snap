// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Idea rendering renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Ideas />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Spinner />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
  height: calc(100vh - 80px);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
",
        ],
      },
      "displayName": "Styled(Container)",
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
      Symbol(Symbol.hasInstance): [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Spinner />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bxivhb",
          "isStatic": true,
          "rules": Array [
            "
  height: calc(100vh - 80px);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
",
          ],
        },
        "displayName": "Styled(Container)",
        "render": [Function],
        "styledComponentId": "sc-bxivhb",
        "target": [Function],
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
        Symbol(Symbol.hasInstance): [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
